src/main/java/io/airlift/compress/lz4/Lz4Compressor.java

```diff
@@ -47,48 +48,55 @@ public int compress(byte[] input, int inputOffset, int inputLength, byte[] outpu
     @Override
     public void compress(ByteBuffer input, ByteBuffer output)
     {
+        // Java 9+ added an overload of various methods in ByteBuffer. When compiling with Java 11+ and targeting Java 8 bytecode
+        // the resulting signatures are invalid for JDK 8, so accesses below result in NoSuchMethodError. Accessing the
+        // methods through the interface class works around the problem
+        // Sidenote: we can't target "javac --release 8" because Unsafe is not available in the signature data for that profile
+        Buffer inputBuffer = input;
```

Comment:
Maybe rename the parameters instead, so that all the local variables don't change